Tasks/AzureRmWebAppDeploymentV4/Tests/WindowsWebAppWebDeployProviderL0Tests.ts

```diff
@@ -0,0 +1,76 @@
+import tl = require('vsts-task-lib');
+import { DeploymentFactory } from '../deploymentProvider/DeploymentFactory';
+import { WindowsWebAppWebDeployProvider } from '../deploymentProvider/WindowsWebAppWebDeployProvider'
+import { IWebAppDeploymentProvider } from '../deploymentProvider/IWebAppDeploymentProvider';
+import { TaskParametersUtility, TaskParameters, DeploymentType } from '../operations/TaskParameters';
+import { stringify } from 'querystring';
+import { PackageType } from 'webdeployment-common/packageUtility';
+import { getMockEndpoint } from '../node_modules/azure-arm-rest/Tests/mock_utils';
+import { mockAzureARMPreDeploymentSteps, mockZipDeploySettings }  from "./mock_utils";
+
+getMockEndpoint();
+mockAzureARMPreDeploymentSteps();
+mockZipDeploySettings();
+
+export class WindowsWebAppWebDeployProviderL0Tests  {
+
+    public static async startWindowsWebAppWebDeployProviderL0Tests() {
+        await WindowsWebAppWebDeployProviderL0Tests.testForPreDeploymentSteps_WebDeployProvider();
+        await WindowsWebAppWebDeployProviderL0Tests.testForPreDeploymentStepsWithSlotEnabled_WebDeployProvider();
+        await WindowsWebAppWebDeployProviderL0Tests.testForUpdateDeploymentStatus_WebDeployProvider();
```

Comment:
Please add for virtual application and when MSDeploy package type

Reply:
Done. In this commit 6799bec05dd4d567b32c2661ea25a6282f03adf3